### PR TITLE
Add matching navigation

### PR DESCRIPTION
**Pull request by qingxin-yuan**
Stream matching screens together; add error handling

### CODE REVIEW

File: js/redux/modules/teams.js [hunk 1]
```diff
@@ -157,6 +154,33 @@ export const fetchOtherMatches = (
...
+      teamSize,
+      users: {
+        [uid]: { fun: score[0], grow: score[1], win: score[2], id: uid }
+      }
+    })
+    .then(async docRef => {
+      // Refetch newly added team, which is a perfect match :)
+      const newTeam = await docRef.get();
+      dispatch(getTeamBestMatch(newTeam.data()));
+    });
```
Comment (siddharthparmar7) [inline js/redux/modules/teams.js]: you should add a catch block to handle errors.

File: js/redux/modules/teams.js [hunk 2]
```diff
@@ -170,6 +194,7 @@ export const addUserToTeam = (score, teamId, userId) => async dispatch => {
       }
     }
   });
+  // add competition id to user
```
Comment (siddharthparmar7) [inline js/redux/modules/teams.js]: unnecessary comment?

File: js/screens/Competition/CompetitionContainer.js [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
-} from "../../redux/modules/teams";
-import { fetchUserWorkstyle } from "../../redux/modules/user";
-import { getUserScore } from "../../redux/modules/user";
+// import {
+//   fetchBestMatch,
+//   fetchOtherMatches,
+//   addUserToTeam
+// } from "../../redux/modules/teams";
+// import { fetchUserWorkstyle } from "../../redux/modules/user";
+// import { getUserScore } from "../../redux/modules/user";
```
Comment (siddharthparmar7) [inline js/screens/Competition/CompetitionContainer.js]: Please remove the code that you do not need.

File: js/screens/JoinTeamModal/JoinTeamModalContainer.js [hunk 4]
```diff
@@ -22,7 +29,10 @@ export default class JoinTeamModalContainer extends Component {
     this.updateFun = this.updateFun.bind(this);
     this.updateGrow = this.updateGrow.bind(this);
     this.updateWin = this.updateWin.bind(this);
-    this.submit = this.submit.bind(this);
+    this.findBestMatch = this.findBestMatch.bind(this);
```
Comment (siddharthparmar7) [inline js/screens/JoinTeamModal/JoinTeamModalContainer.js]: please use arrow functions and remove all the binding from here.

File: js/screens/MatchScreenLoading/MatchScreenLoading.js [hunk 5]
```diff
@@ -9,22 +9,34 @@ import { colors } from "../../config/styles";
...
+              </Text>
+            ) : (
+              <View>
+                <Image
+                  style={{ alignSelf: "center" }}
+                  source={require("../../assets/icons/other/check-mark.png")}
+                />
+                <Text style={styles.title}>{"GROUP FOUND!"}</Text>
+                <Text style={styles.text}>
+                  {"Joining group chat room now…"}
```
Comment (siddharthparmar7) [inline js/screens/MatchScreenLoading/MatchScreenLoading.js]: Please remove curly brackets if you do not need it.

File: js/screens/MatchScreenLoading/MatchScreenLoading.js [hunk 6]
```diff
@@ -37,35 +49,37 @@ const MatchScreenLoading = ({ contentLoading, buttonLoading }) => {
...
+            }}
           />
-        )}
+          {noMatch ? (
+            <Text style={styles.noThanks}>
+              {"No Thanks, I'll come back later"}
+            </Text>
+          ) : (
+            ""
+          )}
```
Comment (siddharthparmar7) [inline js/screens/MatchScreenLoading/MatchScreenLoading.js]: you can write this code little better this way

```
{noMatch && <Text style={styles.noThanks}>"No Thanks, I'll come back later"</Text>}
```

Comment (siddharthparmar7) [reply js/screens/MatchScreenLoading/MatchScreenLoading.js]: from line 66 to 72

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please add dependencies. 
Please make the requested change so I can approve this pull request.

**Comment by qingxin-yuan** [comment]
Changes made!

**Review by siddharthparmar7 — APPROVED** [review]
Thank you!

**Pull request merged**